src/utils/daos.c

```diff
@@ -392,81 +326,90 @@ common_op_parse_hdlr(int argc, char *argv[], struct cmd_args_s *ap)
 		fflush(stderr);
 		return RC_PRINT_HELP;
 	}
-	cmdname = acopy_str(argv[2]);
+	cmdname = strdup(argv[2]);
 
+	/* Parse command options. Use goto on any errors here
+	 * since some options may result in resource allocation.
+	 */
 	while ((rc = getopt_long(argc, argv, "", options, NULL)) != -1) {
 		switch (rc) {
 		case 'G':
-			ap->group = acopy_str(optarg);
+			ap->group = strdup(optarg);
 			break;
 		case 'p':
 			if (uuid_parse(optarg, ap->pool_uuid) != 0) {
 				fprintf(stderr,
 					"failed to parse pool UUID: %s\n",
 					optarg);
-				return RC_NO_HELP;
+				rc = RC_NO_HELP;
+				D_GOTO(out_free, rc);
```

Comment:
you can collapse this (and all other subsequent D_GOTO) into one line:
D_GOTO(out_free, rc = RC_NO_HELP);